core/types.rng

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<grammar ns="http://libero.pub" xmlns="http://relaxng.org/ns/structure/1.0"
+    datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
+
+    <define name="libero.core.types.id">
+
+        <data type="string">
+            <param name="pattern">\s*[A-Za-z0-9\-._]+\s*</param>
```

Comment:
Do we want to set lengths at this level? Is the character range enough?

Reply:
The pattern is for URL-friendliness? `content-store` didn't assume lenghts, just has a technical limit as a constant for the column data type, which could be configured.

Reply:
Can make this wider, the user can always redefine it.